sdk/nodejs/tests/automation/localWorkspace.spec.ts

```diff
@@ -333,17 +332,18 @@ describe("LocalWorkspace", () => {
         };
         const projectName = "inline_node";
         const stackNames = Array.from(Array(10).keys()).map(_ => fullyQualifiedStackName(getTestOrg(), projectName, `int_test${getTestSuffix()}`));
-        const stacks = await Promise.all(stackNames.map(async stackName => LocalWorkspace.createStack({ stackName, projectName, program })));
 
-        const stackConfig: ConfigMap = {
-            "bar": { value: "abc" },
-            "buzz": { value: "secret", secret: true },
-        };
-        await Promise.all(stacks.map(async stack => stack.setAllConfig(stackConfig)));
+        const testStackLifetime = async (stackName: string) => {
+            const stack = await LocalWorkspace.createStack({ stackName, projectName, program });
```

Comment:
Re: #8186 raised by @komalali, this is why I was a bit confused. I think we're doing the right thing here to ensure `--stack` args are used & global state is not.

@justinvp what sort of test are you thinking we'd need to close #8186?

Reply:
The `pulumi stack init` command currently always saves the new stack as the selected stack. This used to cause problems when multiple parallel things were trying to save the file where the current stack is saved, leading to a corrupt file. That has since been fixed so that the file is saved atomically, to avoid corruption.

However, `pulumi stack init` is still saving the current stack, and it really shouldn't be doing that when used from Automation API. If we were going to test it, we could write tests that call automation API's `createStack` and `selectStack` and check the value in the workspace settings hasn't changed.